internal/agent/agent.go

```diff
@@ -70,23 +71,14 @@ func (a *Agent) Run(ctx context.Context) error {
 	}(ctx)
 
 	// create file io writer and reader
-	deviceWriter, deviceReader := initializeFileIO(a.config)
-
-	currentSpecFilePath := filepath.Join(a.config.DataDir, spec.CurrentFile)
-	desiredSpecFilePath := filepath.Join(a.config.DataDir, spec.DesiredFile)
+	deviceReadWriter := fileio.NewReadWriter(fileio.WithTestRootDir(a.config.testRootDir))
```

Comment:
I wonder maybe better to allow providing deviceReadWriter to the object ? In that case there will be no need to for fileio.WithTestRootDir here and test will be able to set which path it wants without setEnv?

Reply:
you mean in the config? that does make sense